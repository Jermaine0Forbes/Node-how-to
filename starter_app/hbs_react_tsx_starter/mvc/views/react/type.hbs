<div class="container">
    <section id="root">
    </section>
</div>
<script src="js/type.js" async></script>