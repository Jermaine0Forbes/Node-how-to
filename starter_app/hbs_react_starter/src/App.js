import React ,{useState, useEffect}from "react";


const App = () => {
    const 
    [color, setColor] = useState("white"),
    [height, setHeight] = useState(50),
    [width, setWidth] = useState(50),
    handleColor = (e) => {
        console.log(e)
        console.log(e.target)
        console.log(e.target.value)
        setColor(e.target.value)
    },
    handleHeight = (e) => {
        setHeight(e.target.value)
    },
    handleWidth = (e) => {
        setWidth(e.target.value)
    };

  


    return (
        <>
           <h2 className="bg-primary text-white px-4"> Hello from  React</h2> 
           <div className="row col-6">
                <form>
                    <div className="form-group">
                        <label className="h3 d-block">Color</label>
                        <select className="form-control" defaultValue={"0"} onChange={(e) => handleColor(e)}>
                            <option value="0" disabled>Choose a color</option>
                            <option value="red">Red</option>
                            <option value="blue">Blue</option>
                            <option value="green">Green</option>
                        </select>
                    </div> 
                    <div className="form-group">
                        <label className="h3 d-block">Height</label>
                        <input type="number" className="form-control" onChange={(e) => handleHeight(e)} defaultValue={height} />
                    </div>
                    <div className="form-group">
                        <label className="h3 d-block">Width</label>
                        <input type="number" className="form-control" onChange={(e) => handleWidth(e)} defaultValue={width}/>
                    </div>
                </form>

           </div>
           <div className="row">
                <ColorBox height={height} width={width} color={color} />
           </div>
        </>
    );
}

export const ColorBox = ({height, width, color}) => {

    return (
        <div id="color-box" className="my-5" style={{height: height+"px", width: width+"px", background: color, transition: "all 0.3s"}}>
        </div>
    )
}

export default App;