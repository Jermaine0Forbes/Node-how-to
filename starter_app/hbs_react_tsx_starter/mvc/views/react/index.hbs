<div class="container">
    <section id="app">
    </section>
    
</div>
<script src="js/app.js" async></script>