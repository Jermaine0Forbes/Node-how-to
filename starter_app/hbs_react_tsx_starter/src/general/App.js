import React ,{useState, useEffect, useReducer, createContext} from "react";
import Context from "./Context";


const App = () => {
    const 
    [color, setColor] = useState("white"),
    [height, setHeight] = useState(50),
    [width, setWidth] = useState(50),
    [food, setFood] = useState("pizza"),
    handleColor = (e) => {
        console.log(e)
        console.log(e.target)
        console.log(e.target.value)
        setColor(e.target.value)
    },
    handleHeight = (e) => {
        setHeight(e.target.value)
    },
    handleWidth = (e) => {
        setWidth(e.target.value)
    },
    handleFood = (e) => {
        setFood(e.target.dataset.food)
    };

  


    return (
        <>
           <h2 className="text-primary "> Hello from  React</h2> 
           <Context/>
           <section>
                <div className="row col-6">
                        <form>
                            <div className="form-group">
                                <label className="h3 d-block" aria-labelledby="color" htmlFor="color">Color</label>
                                <select className="form-control" defaultValue={"0"} name="color" onChange={(e) => handleColor(e)}>
                                    <option value="0" disabled>Choose a color</option>
                                    <option value="red">Red</option>
                                    <option value="blue">Blue</option>
                                    <option value="green">Green</option>
                                </select>
                            </div> 
                            <div className="form-group">
                                <label className="h3 d-block">Height</label>
                                <input type="number" className="form-control" placeholder="height" name="height-inpt" onChange={(e) => handleHeight(e)} defaultValue={height} />
                            </div>
                            <div className="form-group">
                                <label className="h3 d-block">Width</label>
                                <input type="number" className="form-control" onChange={(e) => handleWidth(e)} defaultValue={width}/>
                            </div>
                        </form>

                </div>
                <div className="row">
                        <ColorBox height={height} width={width} color={color} />
                </div>
           </section>
           <section>
                <div className="btn-group mb-4" role="group" aria-label="Basic example">
                    <button type="button" aria-label="pizza-btn" className="btn btn-primary px-5" data-food="pizza" onClickCapture={(e) => {handleFood(e)}} ><i className="fas fa-pizza-slice"></i></button>
                    <button type="button" aria-label="icecream-btn" className="btn btn-warning px-5" data-food="icecream" onClickCapture={(e) => {handleFood(e)}} ><i className="fas fa-ice-cream text-white"></i></button>
                    <button type="button" aria-label="hamburger-btn" className="btn btn-success px-5" data-food="hamburger" onClick={(e) => {handleFood(e)}} ><i className="fas fa-hamburger"></i></button>
                </div>
                <div className="d-flex justify-content-between col-md-5">
                    <ul className=" list-group" aria-label="pizza-menu" style={food == "pizza" ? {display:"block"} :{display:"none"}}>
                        <li className="list-group-item bg-primary text-white">pizza item 1</li>
                        <li className="list-group-item bg-primary text-white">pizza item 2</li>
                    </ul>
                    <ul className="bg-warning list-group" aria-label="icecream-menu" style={food == "icecream" ? {display:"block"} :{display:"none"}}>
                        <li className="list-group-item bg-warning text-white">icecream item 1</li>
                        <li className="list-group-item bg-warning text-white">icecream item 2</li>
                    </ul>
                    <ul className="bg-success list-group" aria-label="hamburger-menu" style={food == "hamburger" ? {display:"block"} :{display:"none"}}>
                        <li className="list-group-item bg-success text-white">hamburger item 1</li>
                        <li className="list-group-item bg-success text-white">hamburger item 2</li>
                    </ul>
                </div>
           </section>
        </>
    );
}




export const ColorBox = ({height, width, color}) => {

    return (
        <div id="color-box" className="my-5" style={{height: height+"px", width: width+"px", background: color, transition: "all 0.3s"}}>
        </div>
    )
}

export const NameForm = () => {

    const 
    [title, setTitle ] = useState(null),
    [first, setFirst ] = useState(null),
    [last, setLast ] = useState(null),
    handleVal = (e) => {
        const val = e.target.value, attr = e.target.getAttribute("aria-label");
        if(val){
            switch(attr)
            {
                case "select-title":
                    setTitle(val)
                    break;
                case "input-first":
                    setFirst(val)
                    break;
                case "input-last":
                    setLast(val)
                    break;
            }
        } 
    };

    return(
        <section>
            <form className="name-form row co-6">
                <div className="form-group">
                    <select defaultValue={0} aria-label="select-title" onChange={(e) => {handleVal(e)}}>
                        <option value="0" disabled>Choose a title</option>
                        <option value="Mr.">Mr.</option>
                        <option value="Ms.">Ms.</option>
                    </select>
                </div>
                <div className="form-group">
                    <input type="text" className="form-control" aria-label="input-first" placeholder="first name"  onChange={(e) => {handleVal(e)}}/>
                </div>
                <div className="form-group">
                <input type="text" className="form-control" aria-label="input-last" placeholder="last name" onChange={(e) => {handleVal(e)}} />
                </div>
                <input type="submit" className="btn btn-primary" value="Submit" onClick={ (e) => handleClick}/>
            </form>
            <div >
                <p>this is a paragraph</p>
                {
                    
                    title && first && last ?
                      <p aria-label="output-name" role="article">Hello, {title} {first} {last}</p>
                    :
                        null
                        
                }
            </div>
        </section>
    )
}

export default App;