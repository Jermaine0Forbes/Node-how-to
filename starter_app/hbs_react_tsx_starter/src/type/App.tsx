import React , {SyntheticEvent, useContext, useRef, useState} from 'react';
import { Button , Box, Container, Grid, TextField,Input } from '@material-ui/core';

const MyTheme = React.createContext(null);

export default function App():JSX.Element
{
    
        const [field , setField] = React.useState(""),
              [data, setData] = useState(""),
                tField = useRef(null),
                handleField = (e:any) => {
                    // console.log(e)
                    setField(e.target.value)
                    
                },
                handleButton = () => {
                    // tField.current.value = "";
                    console.log(tField.current.querySelector("input").value)
                    setData(field)
                    tField.current.querySelector("input").value = "";
                };
    return (
        <MyTheme.Provider  value={data}>
            <Box color="primary.main">
                <Container>
                    <Grid item md={10}>
                        <h1>What is your name?</h1>
                        <Box className="form-group">
                            <TextField id="outlined-basic" onChange={(e) => handleField(e)} className="my-3" ref={tField} label="Enter Name" variant="outlined" placeholder="Enter Name" />
                        </Box>
                        <Button variant="contained" color="primary" onClick={handleButton}>Submit</Button>
                    </Grid>
                    <DisplayName/>
                </Container>
            </Box>
        </MyTheme.Provider>
    )
}

export const DisplayName = () => {
    const contextType = useContext(MyTheme);
    return (
        <Box className="mt-4" aria-label="name-box">
           {contextType}
        </Box>
    )
}