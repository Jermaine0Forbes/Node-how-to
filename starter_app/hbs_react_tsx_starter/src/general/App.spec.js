import React from "react";
import {render, fireEvent, screen, getByLabelText, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import  renderer from "react-test-renderer";
import  App, {ColorBox, NameForm} from "./App";
import { expect } from "@jest/globals";
import 'regenerator-runtime/runtime'


describe('My Test Suite', () => {
  it('should show my first test', () => {
    expect(true).toEqual(true);
  });
});

describe("ColorBox component", () => {
  it('snapshot renders', () => {
    const component = renderer.create(<ColorBox height={80} width={80} color="sienna" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


})

describe("App has been rendered", () => {

  it("should see the title in the component", () => {
    
      // const {getByText, getByLabelText} = render(<App/>);

      render(<App/>);

    //  const header = screen.getByRole("heading", {name:/Hello from React/i});
    const header = screen.getByText(/Hello from React/i);
    //  const header = screen.getByText("Height");
    //  const header = screen.getByLabelText('Color'); 
    expect(header).toBeInTheDocument();
  })

  it(" should change the size of the color box", () => {
    render(<App />)
    // const input =  screen.getByLabelText("height-inpt");
    const input =  screen.getByPlaceholderText("height");
    fireEvent.change(input,{target:{value:80}});
    const colorBox = document.getElementById("color-box");
    
    expect(colorBox.style.height).toBe("80px");
    // expect(input.value).toBe("80");
  })

  it("should display a new food menu when you click on a different food icon and hide the previous food menu", () => {

    // const {debug, container} = render(<App/>)
    render(<App/>)
    // debug();
    const btn = screen.getByRole("button",{name:/icecream-btn/i})
    const currentFoodMenu = screen.getByRole("list",{name:/pizza-menu/i})
    fireEvent.click(btn)
    const newFoodMenu = screen.getByRole("list",{name:/icecream-menu/i});
    expect(btn).toHaveAttribute("data-food","icecream")
    expect(currentFoodMenu).toBeInTheDocument();
    expect(newFoodMenu).toBeVisible();
    expect(currentFoodMenu).not.toBeVisible();

  })

  it("should have a button with a specific class", () => {
    render(<App/>);

    const btn = screen.getByRole("button", {name:/icecream-btn/i})
    expect(btn).toHaveClass("btn-warning")
  });



})

describe("NameForm component", () => {
  it(" should the display the name and title", async () =>  {
     const {debug, container} = render(<NameForm/>)

     const 
        title = screen.getByRole("combobox", {name:/select-title/i}),
        first = screen.getByRole("textbox", {name:/input-first/i}),
        last = screen.getByRole("textbox", {name:/input-last/i});

            // fireEvent.change(title, {target:{value:"Mr."}});
            userEvent.selectOptions(title, "Mr.");
            // fireEvent.change(first, {target:{value:"Jermaine"}});
            userEvent.type(first, "Jermaine");
            // fireEvent.change(last, {target:{value:"Forbes"}});
            userEvent.type(last, "Forbes");

            const output =  container.querySelector("p[aria-label='output-name']");
            expect(output).toBeInTheDocument();
            expect(output.textContent).toBe("Hello, Mr. Jermaine Forbes");
  });


})