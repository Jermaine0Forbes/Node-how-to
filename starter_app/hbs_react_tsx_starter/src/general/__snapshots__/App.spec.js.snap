// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox component snapshot renders 1`] = `
<div
  className="my-5"
  id="color-box"
  style={
    Object {
      "background": "sienna",
      "height": "80px",
      "transition": "all 0.3s",
      "width": "80px",
    }
  }
/>
`;
